### ISSUE TITLE
Bug: (sort-variable-declarations) Destructured assignments dependencies are not detected

### ISSUE THREAD

**Issue by hugop95** [azat-io/eslint-plugin-perfectionist#594]
### Describe the bug

Destructured assignments dependencies are not detected and will enforce code that may not compile.

### Code example

```ts
let a = "a",
  [{
    b = a,
  }] = {}
```

Will be sorted to

```ts
let [{
    b = a,
  }] = {},
  a = "a"
```

which will not compile (`TS2448: Block-scoped variable a used before its declaration.`).

### ESLint version

8.45.0+

### ESLint Plugin Perfectionist version

4.15.0

### Validations

- [x] Read the [docs](https://perfectionist.dev).
- [x] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
Should we release the fix now?

**Comment by hugop95** [comment]
@azat-io Sure, I don't have other fixes in mind.

**Comment by azat-io** [comment]
Fixed in 1def9db2e75c81893cb1db72c1821064b8080dd8.
Released in [v4.15.1](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v4.15.1).
